### PR TITLE
Make sure loss function goes down always

### PR DESCRIPTION
**Pull request by jasmainak**
@pavanramkumar do you agree this is a useful sanity check?

### CODE REVIEW

File: tests/test_pyglmnet.py [hunk 1]
```diff
@@ -164,6 +164,7 @@ def test_glmnet():
 
             X_train = scaler.fit_transform(X_train)
             glm.fit(X_train, y_train)
+            assert_true(np.all(np.diff(glm._loss) < 0))
```
Comment (jasmainak) [inline tests/test_pyglmnet.py]: @pavanramkumar this test passes for all the distributions of the solver 'batch-gradient', but it does not pass for 'cdfast'

File: pyglmnet/pyglmnet.py [hunk 2]
```diff
@@ -676,6 +676,9 @@ def fit(self, X, y):
         self.beta0_ = beta[0]
         self.beta_ = beta[1:]
         self.ynull_ = np.mean(y)
+
+        # Update loss
+        self._loss = L
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: @pavanramkumar I profiled the code and it seems computing the loss function takes up 1/4th of the time. You could simply use the norm of the gradient for stopping criteria and make it more efficient.

**Comment by jasmainak** [comment]
superseded by https://github.com/glm-tools/pyglmnet/pull/242
